hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/storage/HoodieSparkParquetReader.java

```diff
@@ -79,8 +82,8 @@ public BloomFilter readBloomFilter() {
   }
 
   @Override
-  public Set<String> filterRowKeys(Set<String> candidateRowKeys) {
-    return parquetUtils.filterRowKeys(conf, path, candidateRowKeys);
+  public Set<String> filterRowKeys(Option<BaseKeyGenerator> keyGeneratorOpt, Option<SerializableSchema> schemaOpt, Set<String> candidateRowKeys) {
```

Comment:
Chatted offline: we should avoid exposing FileReader and ParquetUtils to the KeyGenerators, instead we need to think about providing an abstraction providing:

 - Proper projected schema necessary for it to be able to produce the record key
 - Produce actual key (from the projected version of the object)

This abstraction have to be
 - As performant as using meta-fields for Simple KG